### PR TITLE
Fix remaining types

### PR DESCRIPTION
**Pull request by Solpatium**
Only `typed_data.py` has global `reportGeneralTypeIssues` enabled now. It will be addressed after logic fixes.

### CODE REVIEW

File: starknet_py/contract.py [hunk 1]
```diff
@@ -99,8 +98,9 @@ async def wait_for_acceptance(
 @add_sync_methods
 @dataclass(frozen=True)
 class InvokeResult(SentTransaction):
-    contract: ContractData = None
-    invoke_transaction: InvokeFunction = None
+    # We ensure these not None in __post_init__
```
Comment (cptartur) [inline starknet_py/contract.py:101]:
```suggestion
    # We ensure these are not None in __post_init__
```

File: starknet_py/contract.py [hunk 2]
```diff
@@ -99,8 +98,9 @@ async def wait_for_acceptance(
...
+    contract: ContractData = None  # pyright: ignore
+    invoke_transaction: InvokeFunction = None  # pyright: ignore
```
Comment (cptartur) [inline starknet_py/contract.py:103]: This is not validated in `__post_init__`

File: starknet_py/contract.py [hunk 3]
```diff
@@ -130,7 +131,7 @@ def __init__(
         client: Client,
         payload_transformer: FunctionCallSerializer,
         contract_data: ContractData,
-        max_fee: int,
+        max_fee: Optional[int],
```
Comment (cptartur) [inline starknet_py/contract.py:134]: Why was this changed?

File: starknet_py/net/gateway_client.py [hunk 4]
```diff
@@ -396,14 +394,14 @@ def get_block_identifier(
...
+            "entry_point_selector": hex(tx.selector),
             "calldata": [str(i) for i in tx.calldata],
         }
     return {
-        "contract_address": hex(tx.to_addr),
-        "entry_point_selector": hex(tx.selector),
+        "contract_address": hex(tx.contract_address),
+        "entry_point_selector": hex(tx.entry_point_selector),
         "calldata": [str(i) for i in tx.calldata],
     }
```
Comment (cptartur) [inline starknet_py/net/gateway_client.py:407]: I wonder why it only works in that order. 🤔

File: starknet_py/utils/iterable.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from typing import TypeVar, Iterable, Union
+
+T = TypeVar("T")
+
+# pyright: reportGeneralTypeIssues=false
+def ensure_iterable(value: Union[T, Iterable[T]]) -> Iterable[T]:
```
Comment (cptartur) [inline starknet_py/utils/iterable.py:6]: `Union[T, Iterable[T]]` it would make more sense to type it as `Any` because it is basically that.

**Review by cptartur — CHANGES_REQUESTED** [review]

File: starknet_py/contract.py (hunk 3 above)
Comment (Solpatium) [reply starknet_py/contract.py:134]: Because in fact it accepts it.

File: starknet_py/utils/iterable.py (hunk 5 above)
Comment (Solpatium) [reply starknet_py/utils/iterable.py:6]: How would you type the return value without losing type in that case without providing generic type directly?

File: starknet_py/contract.py (hunk 2 above)
Comment (Solpatium) [reply starknet_py/contract.py:103]: Yes is is
```
    def __post_init__(self):
        assert self.contract is not None
        assert self.invoke_transaction is not None
```

File: starknet_py/contract.py (hunk 3 above)
Comment (Solpatium) [reply starknet_py/contract.py:134]: And code depends on it

File: starknet_py/utils/iterable.py (hunk 5 above)
Comment (cptartur) [reply starknet_py/utils/iterable.py:6]: I though maybe we could still use `Iterable[T]` as return type. In that case, we can keep it as it is.

**Review by cptartur — APPROVED** [review]

File: starknet_py/utils/iterable.py (hunk 5 above)
Comment (Solpatium) [reply starknet_py/utils/iterable.py:6]: Python would have no way of knowing what the T should be there

**Review by war-in — APPROVED** [review]
